index.js

```diff
@@ -67,18 +87,26 @@ function sanitize(target, options) {
 
     return {
       shouldRecurse: shouldRecurse,
-      key: key
+      key: key,
     };
   });
 
   return target;
 }
 
 function middleware(options) {
-  return function(req, res, next) {
-    ['body', 'params', 'headers', 'query'].forEach(function(k) {
-      if(req[k]) {
-        req[k] = sanitize(req[k], options);
+  return function (req, res, next) {
+    ['body', 'params', 'headers', 'query'].forEach(function (k) {
+      if (req[k]) {
+        req[k] = sanitize(
+          req[k],
+          options,
+          options
+            ? options.allowDots
+              ? TEST_REGEX_WITHOUT_DOT
+              : TEST_REGEX
```

Comment:
There are 3 places in the codebase now that return the regex depending on the `options.allowDots` property. I think this merits a separate function, `getTestRegex` which could return the correct regex. This will hopefully reduce the congitive load when reading the code.

``` js
function getTestRegex(options = {}) {
  return options.allowDots ? TEST_REGEX_WITHOUT_DOT : TEST_REGEX;
}
```

This function can then be called wherever the code needs to decide on the regex that is used.